File: app/src/main/java/com/carpinelli/testeandroid/ui/main/MainActivity.java
package com.carpinelli.testeandroid.ui.main;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.carpinelli.testeandroid.R;

public class MainActivity extends AppCompatActivity {

    private TabLayout tabLayout;
    private ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewPager = findViewById(R.id.viewPager);
        viewPager.setAdapter(new PageViewAdapter(getSupportFragmentManager()));

        tabLayout = findViewById(R.id.tabLayout);
        tabLayout.setTabTextColors(ContextCompat.getColor(this,R.color.colorPrimaryDark), ContextCompat.getColor(this,R.color.colorPrimary));
        tabLayout.addTab(tabLayout.newTab().setText(getResources().getText(R.string.invest)));
        tabLayout.addTab(tabLayout.newTab().setText(getResources().getText(R.string.form)));

    }


    public class PageViewAdapter extends FragmentPagerAdapter {

        public PageViewAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            switch (position){
                case 0:
                    return new InvestFragment();
                case 1:
                    return new FormFragment();
            }
            return new Fragment();
        }

        @Override
        public int getCount() {
            return 2;
        }
    }

}

File: app/src/main/java/com/anabhomasi/androidapp/views/activities/MainActivity.kt
package com.anabhomasi.androidapp.views.activities

import android.graphics.drawable.Drawable
import android.os.Bundle
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.content.ContextCompat
import android.support.v4.view.ViewPager.OnPageChangeListener
import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import com.anabhomasi.androidapp.R
import com.anabhomasi.androidapp.views.adapters.PageAdapter
import com.anabhomasi.androidapp.views.fragments.DynamicFormFragment
import com.anabhomasi.androidapp.views.fragments.FormFragment
import com.anabhomasi.androidapp.views.fragments.FundFragment
import com.anabhomasi.androidapp.views.fragments.SuccessFormFragment
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(),
        DynamicFormFragment.OnFragmentInteractionListener,
        FundFragment.OnFragmentInteractionListener,
        SuccessFormFragment.OnFragmentInteractionListener{

    override fun onFragmentInteraction(page: Int) {

        myViewPager.currentItem = page
    }

    private var adapterViewPager: FragmentPagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        adapterViewPager = PageAdapter(supportFragmentManager)
        myViewPager.adapter = adapterViewPager

        btnContato.setOnClickListener {
            myViewPager.currentItem = 1
        }

        btnInvestimento.setOnClickListener {
            myViewPager.currentItem = 0
        }

        toolbar_title.text = myViewPager.adapter?.getPageTitle(myViewPager.currentItem)
        adjustButtonColorStates(myViewPager.currentItem)

        // Attach the page change listener inside the activity
        myViewPager.addOnPageChangeListener(object : OnPageChangeListener {

            // This method will be invoked when a new page becomes selected.
            override fun onPageSelected(position: Int) {
                toolbar_title.text = myViewPager.adapter?.getPageTitle(position)
                adjustButtonColorStates(position)
            }

            // This method will be invoked when the current page is scrolled
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
                // Code goes here
            }

            // Called when the scroll state changes:
            // SCROLL_STATE_IDLE, SCROLL_STATE_DRAGGING, SCROLL_STATE_SETTLING
            override fun onPageScrollStateChanged(state: Int) {
                // Code goes here
            }
        })

    }
    private fun adjustButtonColorStates(position: Int) {
        when(position){
            0 -> {
                btnInvestimento.background = selectedState()
                btnContato.background = unselectedState()
            }
            1 -> {
                btnInvestimento.background = unselectedState()
                btnContato.background = selectedState()
            }
        }
    }

    private fun selectedState() : Drawable?{
        return ContextCompat.getDrawable(this, R.color.colorPrimaryDark)
    }

    private fun unselectedState() : Drawable?{
        return ContextCompat.getDrawable(this, R.drawable.tab_unselected_color_state)
    }

}